### PR TITLE
py-horovod: new versions

### PR DESCRIPTION
**Pull request by aweits**
@adamjstewart @tgaddair

### CODE REVIEW

**Comment by tgaddair** [comment]
Thanks @aweits!  Just a heads up, we'll also be releasing v0.21.3 today, which has a bugfix for TensorFlow 2.4 users:

https://github.com/horovod/horovod/pull/2667

**Comment by aweits** [comment]
> Thanks @aweits! Just a heads up, we'll also be releasing v0.21.3 today, which has a bugfix for TensorFlow 2.4 users:
> 
> [horovod/horovod#2667](https://github.com/horovod/horovod/pull/2667)

Cool! Let's hold off on merging this until that's tagged then...

**Comment by tgaddair** [comment]
Thanks @aweits, v0.21.3 is now released.

**Comment by aweits** [comment]
> Thanks @aweits, v0.21.3 is now released.

Builds, lightly tested against just the tensorflow framework (tensorflow@2.4.0), using tensorflow2_synthetic_benchmark.py on a standalone single-gpu machine.

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Required version of py-petastorm has changed, py-h5py should probably be :2.999, not 2.9:. I forget where the required versions of TF/PyTorch/mxnet/etc are documented.

File: var/spack/repos/builtin/packages/py-horovod/package.py [hunk 1]
```diff
@@ -69,12 +69,14 @@ class PyHorovod(PythonPackage, CudaPackage):
     depends_on('py-cffi@1.4.0:',        type=('build', 'run'),         when='frameworks=pytorch')
     depends_on('mxnet@1.4.1:+python',   type=('build', 'link', 'run'), when='frameworks=mxnet')
     depends_on('py-keras@2.0.8,2.1.2:', type=('build', 'run'),         when='frameworks=keras')
-    depends_on('py-h5py@2.9:',          type=('build', 'run'),         when='frameworks=spark')
+    depends_on('py-h5py@2.999:',        type=('build', 'run'),         when='frameworks=spark')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-horovod/package.py]:
```suggestion
    depends_on('py-h5py@:2.999',        type=('build', 'run'),         when='frameworks=spark')
```

File: var/spack/repos/builtin/packages/py-petastorm/package.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ class PyPetastorm(PythonPackage):
     pypi = "petastorm/petastorm-0.8.2.tar.gz"
 
     maintainers = ['adamjstewart']
-
+    version('0.9.8', sha256='66009b7ad3f08b0485a748f12b2095a0d2470e04f0c63de43cd5b099f270c268')
     version('0.8.2', sha256='7782c315e1ee8d15c7741e3eea41e77b9efce661cf58aa0220a801db64f52f91')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-petastorm/package.py:17]: Can you also add:
```python
depends_on('python@3:', when='@0.9.8:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**